docs/source/usage_guides/deepspeed_multiple_model.md

```diff
@@ -0,0 +1,215 @@
+
+Normally, you would use a single [`utils.DeepSpeedPlugin`] at a time, however in this case we have two seperate configurations. Accelerate allows you to create and use multiple plugins **if and only if** they are in a `dict` so that we can reference/enable the proper plugin when needed:
+
+```python
+from accelerate.utils import DeepSpeedPlugin
+
+zero2_plugin = DeepSpeedPlugin(hf_ds_config="zero2_config.json")
+zero3_plugin = DeepSpeedPlugin(hf_ds_config="zero3_config.json")
+
+deepspeed_plugins = {"student": zero2_plugin, "teacher": zero3_plugin}
+```
+
+When doing so, `zero2_config.json` should be configured for full training (so specifying `scheduler` and `optimizer` if not utilizing your own) while `zero3_config.json` should *only* be configured for the model,
+such as:
+
+```json
+{
+    "bf16": {
+        "enabled": "auto"
+    },
+    "zero_optimization": {
+        "stage": 3,
+        "offload_param": {
+            "device": "cpu",
+            "pin_memory": true
+        },
+        "overlap_comm": true,
+        "sub_group_size": 1e9,
+        "reduce_bucket_size": 1e9,
+        "stage3_prefetch_bucket_size": 1e9,
+        "stage3_param_persistence_threshold": 1e9,
+        "stage3_max_live_parameters": 1e9,
+        "stage3_max_reuse_distance": 1e9,
+        "stage3_gather_16bit_weights_on_model_save": true
+    },
+    "train_micro_batch_size_per_gpu": 1
+}
+```
+
+<Tip>
+
+    DeepSpeed will raise an error if we don't specify a `train_micro_batch_size_per_gpu`, despite not really 
+    training this particular model
+
+</Tip>
+
+From here, we can create a single [`Accelerator`] and pass in both configurations:
+
+```python
+from accelerate import Accelerator
+
+accelerator = Accelerator(deepspeed_plugin=deepspeed_plugins)
+```
+
+Now let's see how to use them.
+
+### The student model
+
+By default, `accelerate` will set the first item in the `dict` as the default/enabled plugin (so here our `"student"` plugin). We can verify this by using the [`utils.deepspeed.get_active_deepspeed_plugin]` function to see which one is being used:
+
+```python
+active_plugin = get_active_deepspeed_plugin(accelerator.state)
+assert active_plugin is deepspeed_plugins["student"]
+```
+
+Also the `AcceleratorState` will keep the active DeepSpeed plugin saved in `state.deepspeed_plugin`:
+```python
+assert active_plugin is accelerator.deepspeed_plugin
+```
+
+Since the `student` is the currently active plugin, let's go ahead and prepare our model, optimizer, and scheduler like normal:
+
+```python
+student_model, optimizer, scheduler = ...
+student_model, optimizer, scheduler, train_dataloader = accelerator.prepare(student_model, optimizer, scheduler, train_dataloader)
+```
+
+From here, we need to deal with the teacher model.
+
+### The teacher model
+
+First we need to tell the `Accelerator` that the zero3/teacher ZeRO configuration should be used:
+
+```python
+accelerator.state.enable_deepspeed_plugin("teacher")
```

Comment:
yeah, now reading the demo, I think I agree with the earlier discussion - `enable` is a bad terminology for something that requires only 1 of many state. Enable implies that more than one thing can be enabled at the same time, but this is not the case.

Toggle could work for 2 objects. But then we may have more than 2 models, in which case toggle doesn't fit either.

So probably `select` is the right verb here, as it implies one among many.

But yet you did say in the docs that only one plugin can be enabled at any given time... not sure. I just think select is more self-documenting.

Reply:
Will move forward with `.select()` then. If we get to a point where for some reason we have multiple active, we can deprecate and change it back (or just add)